Documentation review (#3)
eizamaliev/open_model_zoo: CONTRIBUTING.md

# How to Contribute Models to Open Model Zoo

We appreciate your intention to contribute model to the OpenVINO&trade; Open Model Zoo (OMZ). OMZ is licensed under the Apache\* License, Version 2.0. By contributing to the project, you agree to the license and copyright terms therein and release your contribution under these terms. Note that we accept models under permissive licenses, such as **MIT**, **Apache 2.0**, and **BSD-3-Clause**. Otherwise, it might take longer time to get your model approved.

Frameworks supported by the Open Model Zoo: 
* Caffe\*
* Caffe2\* (via conversion to ONNX\*)
* TensorFlow\*
* PyTorch\* (via conversion to ONNX\*)
* MXNet\*

## Pull Request Requirements

To contribute to OMZ, create a pull request (PR) in this repository using the `develop` branch.
Pull requests are strictly formalized and are reviewed by the OMZ maintainers for consistence and legal compliance.

> Each PR must contain:

Reviewer: `Each` is very strong condition, because someone want just add demo or fix minor issue. This guide is *only* about model contribution.

demo file|`demos`

### Tests

Your PR must pass next tests:
* Model is downloadable by the `tools/downloader/downloader.py` script. See [Configuration file](#configuration-file) for details.
* Model is convertible by the `tools/downloader/converter.py` script. See [Model conversion](#model-conversion) for details.
* Model is usable by demo or sample and provides adequate results. See [Demo](#demo) for details.
* Model passes accuracy validation. See [Accuracy validation](#accuracy-validation) for details.

### PR Rejection

Your PR may be rejected in some cases, for example:
* If a license is inappropriate (such as GPL-like licenses).
* If a dataset is inaccessible.
* If the PR fails one of the tests above.

## Configuration File

The model configuration file contains information about model: what it is, how to download it, and how to convert it to the IR format. This information must be specified in the `model.yml` file that must be located in the model subfolder. 

Refer to the detailed descriptions of each file provided below.

**`description`**

Description of the model. Must match with the description from the model [documentation](#documentation).

**`task_type`**

> Model task class, see [Model information dumper usage](tools/downloader/README.md#model-information-dumper-usage) for details. If there is no task class of your model, add a new one to the list `KNOWN_TASK_TYPES` of the [tools/downloader/common.py](tools/downloader/common.py) file.

Reviewer: > see [Model information dumper usage](tools/downloader/README.md#model-information-dumper-usage)

it is better to use `here` or `in this file`, or whatever, but not `Model information dumper usage`

```suggestion
Model task class ( you can find list of task [here](tools/downloader/README.md#model-information-dumper-usage) in `task_type` element.
```

Author: It's the name of the file. It's better to tell user where we're pointing them to.

Reviewer: I understand, but not in this case

Author: I guess I'll just make the _Model task class_ itself a link

Reviewer: It is not well too. The point is that the link is not directly to model classes, user need to find them on this page. So, I prefer less explicit link description, like `for details`, or `additional info`.

*You can try it by yourself: open the link and seek `task_type` list element. That are model classes.*

> **NOTE 1**: Image preprocessing parameters (mean and scale) must be built into a converted model to simplify model usage.

> **NOTE 2**: If a model input is a color image, color channel order should be `BGR`.

*After this step you get **conversion parameters** for the Model Optimizer.*

## Demo

A demo shows the main idea of how to infer a model using IE. If your model solves one of the tasks supported by the Open Model Zoo, try to find an appropriate option from [demos](https://docs.openvinotoolkit.org/latest/_demos_README.html) or [samples](https://docs.openvinotoolkit.org/latest/_docs_IE_DG_Samples_Overview.html). Otherwise, you must provide your own demo (C++ or Python).

Demos are required to support the following keys:

Keys | Explanation
--|--
  `-i "<input>"`      |       Required. Input to process.
   `-m "<path>"`      |        Required. Path to an .xml file with a trained model. If the demo uses several models at the same time, use other keys prefixed with `-m`.
   `-d "<device>"`    |        Optional. Default is CPU.
   `-no_show`          |       Optional. Do not visualize inference results.

> 

Reviewer: The table does not fit well in this situation. List is preferable.
